tmt/steps/prepare/install.py

```diff
@@ -156,19 +156,26 @@ def go(self, guest):
         if self.opt('dry'):
             return
 
-        # Prepare the right dnf/yum command
+        # Prepare the applicable package manager command
         self.debug('Check if sudo is necessary.', level=2)
         user = guest.execute('whoami')[0].strip()
         sudo = '' if user == 'root' else 'sudo '
-        self.debug('Check if dnf is available.', level=2)
-        skip = ' --skip-broken' if self.get('missing') == 'skip' else ''
         try:
-            guest.execute('rpm -q dnf')
-            command = f"{sudo}dnf{skip}"
+            self.debug('Check for rpm-ostree.', level=2)
+            skip = True if self.get('missing') == 'skip' else False
+            guest.execute('rpm -q rpm-ostree')
```

Comment:
I think you want to check if you are on a ostree-based system, rather than test for the presence of the `rpm-ostree` RPM.  (It is possible to install `rpm-ostree` in a container or in something like vanilla Fedora).

```suggestion
            guest.execute('stat /run/ostree-booted')
```

Reply:
I was also going to ask if there is a better way how to detect we're running on an ostree-based distro. Thanks much for this very useful hint!

Reply:
Yes! this is awesome Thank you!